agent/ecscni/plugin.go

```
@@ -0,0 +1,201 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package ecscni
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/cihub/seelog"
+	"github.com/containernetworking/cni/libcni"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/pkg/errors"
+)
+
+// NewClient creates a client of ecscni which is used to invoke the plugin
+func NewClient(cfg *Config) CNIClient {
+	pluginPath := CNIPath
+	if cfg.PluginPath != "" {
+		pluginPath = cfg.PluginPath
+	}
+
+	cniVersion := CNIVersion
+	if cfg.CNIVersion != "" {
+		cniVersion = cfg.CNIVersion
+	}
+
+	libcniConfig := &libcni.CNIConfig{
+		Path: []string{pluginPath},
+	}
+
+	return &cniClient{
+		pluginPath: pluginPath,
+		cniVersion: cniVersion,
+		subnet:     ECSSubnet,
+		libcni:     libcniConfig,
+	}
+}
+
+// SetupNS will set up the namespace of container, including create the bridge
+// and the veth pair, move the eni to container namespace, setup the routes
+func (client *cniClient) SetupNS(cfg *Config) error {
+	vethName := defaultEthName
```

There should be no `defaultEthName`. If unspecified, we should return an error.